operator/apis/loki/v1/lokistack_types.go

```diff
@@ -791,6 +791,112 @@ type IngestionLimitSpec struct {
 	PerStreamRateLimitBurst int32 `json:"perStreamRateLimitBurst,omitempty"`
 }
 
+// OTLPAttributeAction defines the action to executed when indexing
+// OTLP resource attributes. Resource attributes can be either added
+// to the index, the chunk structured metadata or enterily dropped.
+//
+// +kubebuilder:validation:Enum=index_label;structured_metadata;drop
+type OTLPAttributeAction string
+
+const (
+	// IndexLabelAction stores a Resource Attribute as a label in index to identify streams.
+	IndexLabelAction OTLPAttributeAction = "index_label"
+	// StructuredMetadataAction stores an Attribute as Structured Metadata with each log entry.
+	StructuredMetadataAction OTLPAttributeAction = "structured_metadata"
+	// DropAction drops Attributes for which the Attribute name does match the regex.
+	DropAction OTLPAttributeAction = "drop"
+)
+
+// OTLPAttributesSpec contains the configuration for a set of attributes
+// to store them as index labels or Structured Metadata or drop them altogether.
+type OTLPAttributesSpec struct {
+	// Action defines the indexing action for the selected attributes.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
```

Comment:
Is empty ("") a valid value for this attribute? If not I'd suggest we make it `required`.